src/lib/hyrise.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "concurrency/transaction_manager.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/plugin_manager.hpp"
+#include "utils/singleton.hpp"
+
+namespace opossum {
+
+class Hyrise : public Singleton<Hyrise> {
+ public:
+  static void reset() { get() = Hyrise{}; }
+
+  PluginManager plugin_manager;
+  StorageManager storage_manager;
+  TransactionManager transaction_manager;
+};
```

Comment:
I just noticed that `CurrentScheduler` also implements a singleton. I guess we should include it here as well?

Reply:
No, it just holds static methods. In fact, it should be changed so that it is not even initializable.

Reply:
True, that singleton stuff should be removed then.

Reply:
Actually, `CurrentScheduler` does not really implement the Singleton pattern. It does not create any instances, which is a key property of the pattern. Instead, it just holds an instance of `AbstractScheduler`, not an instance of `CurrentScheduler`. The following comment is misleading:

https://github.com/hyrise/hyrise/blob/f452c8c71ebb200e9b1b224195e0e669ba279442/src/lib/scheduler/current_scheduler.hpp#L14-L20


Anyhow, in my opinion the current scheduler instance belongs to the *global state* of Hyrise. Or am I wrong?

Reply:
Should have referenced #1786 here.

Reply:
I'd make this a follow-up so that we can get this in somewhat quickly. Considering that this touches 137 files, I'd like to keep the number of conflicts low.